xbmc/addons/Addon.cpp

```diff
@@ -350,16 +400,42 @@ bool CAddon::SettingsToXML(CXBMCTinyXML &doc) const
   return true;
 }
 
-std::shared_ptr<CAddonSettings> CAddon::GetSettings()
+bool CAddon::InitSettings(uint32_t instance)
 {
   // initialize addon settings if necessary
-  if (m_settings == nullptr)
+  if (m_settings.find(instance) == m_settings.end() || !m_settings[instance].m_control)
   {
-    m_settings = std::make_shared<CAddonSettings>(enable_shared_from_this::shared_from_this());
-    LoadSettings(false);
+    CSettingsData data;
+
+    data.m_control = std::make_shared<CAddonSettings>(enable_shared_from_this::shared_from_this());
+    if (instance == ADDON_SETTINGS_ID)
+    {
+      data.m_baseSettingsPath =
+          URIUtils::AddFileToFolder(m_addonInfo->Path(), "resources", "settings.xml");
+      data.m_userSettingsPath = URIUtils::AddFileToFolder(m_profilePath, "settings.xml");
+    }
+    else
+    {
+      data.m_baseSettingsPath =
+          URIUtils::AddFileToFolder(m_addonInfo->Path(), "resources", "instance-settings.xml");
+      data.m_userSettingsPath = URIUtils::AddFileToFolder(
+          m_profilePath, StringUtils::Format("instance-settings-{}.xml", instance));
+    }
+
+    m_settings[instance] = std::move(data);
+    return true;
   }
 
-  return m_settings;
+  return false;
+}
+
+
+std::shared_ptr<CAddonSettings> CAddon::GetSettings(uint32_t instance)
```

Comment:
could this method be const?

Reply:
Does not works direct, by the `InitSettings` and `LoadSettings` become it a bit in conflict.